lib/project_types/extension/features/argo_cli_compatibility.rb

```diff
@@ -0,0 +1,46 @@
+module Extension
+  module Features
+    class ArgoCliCompatibility
+      include SmartProperties
+
+      property! :context, accepts: ShopifyCli::Context
+      property! :renderer_package, accepts: Features::ArgoRendererPackage
+
+      ARGO_ADMIN_CLI = "@shopify/argo-admin-cli"
+      SUPPORTED_AFTER_VERSION = "0.9.3"
+
+      def backwards_compatible?(package_name:, version:)
+        js_system = ShopifyCli::JsSystem.new(ctx: context)
+        Tasks::FindNpmPackages
+          .at_least_one_of(package_name, js_system: js_system)
+          .unwrap { |_e| context.abort(context.message("features.argo.package_not_found", package_name)) }
+          .first
+          .tap do |package|
+            return Gem::Version.new(package.version) < Gem::Version.new(version)
```

Comment:
I'm not 100% sure, but you might need to use `::Gem::Version` instead of `Gem::Version`.

I just tried to run `bin/shopify serve` from this branch and got the following error:

```
/Users/joey/src/github.com/Shopify/shopify-app-cli/lib/project_types/extension/features/argo_cli_compatibility.rb:19:in `block in backwards_compatible?': 
uninitialized constant Extension::Features::ArgoCliCompatibility::Gem (NameError)
```

Reply:
No, I don't think it's an issue with how you're running the code, someone reported a [similar issue](https://github.com/Shopify/shopify-app-cli/issues/1196) with an older version of this (which has since been removed), but neither I or @t6d have been able to reproduce. Definitely worth looking into!